client.go

```diff
@@ -151,6 +152,10 @@ func (client *client) Close() error {
 		return ErrClosedClient
 	}
 
+	// shutdown and wait for the background thread before we take the lock, to avoid races
+	close(client.closer)
+	<-client.closed
```

Comment:
will this deadlock if the [RefreshFrequency](https://github.com/Shopify/sarama/blob/d77967306da5ac9363067b15e6c0f217cfe317e3/client.go#L533) is zero?